src/MedPrep.Api/Controllers/VideoUploadController.cs

```diff
@@ -40,4 +40,95 @@ public async Task<ActionResult<VideoUploadResponse>> UploadVideo([FromForm] Vide
 
         return this.Ok(response);
     }
+
+    [HttpGet]
+    [Route("id")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<VideoRequestResponse>> GetVideoById([FromRoute] Guid id)
+    {
+        var video = await this.videoService.FetchVideoInfo(id);
+
+        if (video == null)
+        {
+            return this.NotFound();
+        }
+
+        // Map service response to VideoResponse DTO
+        var videoResponse = new VideoRequestResponse
+        {
+            Title = video.Title,
+            Description = video.Description,
+            NextVideo = video.NextVideo,
+            PrevVideo = video.PrevVideo,
+            VideoSource = video.VideoSource,
+            SubtitleSource = video.SubtitleSource,
+            CourseModule = video.CourseModule != null ? new CourseModuleDto
+            {
+                Id = video.CourseModule.Id,
+                Name = video.CourseModule.Name
+            } : null,
+            Playlist = video.Playlist != null ? new PlaylistDto
+            {
+                Id = video.Playlist.Id,
+                Name = video.Playlist.Name
+            } : null
+        };
+
+        return this.Ok(videoResponse);
+    }
+
+    [HttpGet]
+    [Route("play/id")]
```

Comment:
To make it more concise you can just do this and get rid of the Route attribute entirely
```csharp
[HttpGet("play/{id}")]
```